File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> A, vector<int> B)
{
    sort(A.begin(), A.end());
    sort(B.begin(), B.end(), greater<int>());
    for(int a : A)
    {
        cout << a << " ";
    }
    cout << "\n";
    for(int b : B)
    {
        cout << b << " ";
    }

    int result = 0;
    for(int i = 0; i<A.size(); i++)
    {
        result += A[i] * B[i];   
    }
    

    return result;
}

File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.cs
using System;

public class Solution 
{
    public int solution(int[] A, int[] B) 
    {
        Array.Sort(A);
        Array.Sort(B);
        int min = 0;
        for(int i=0; i<A.Length; i++)
        {
            min += A[i] * B[A.Length - i - 1];
        }
        return min;
    }   
}